### PR TITLE
{tools}[GCCcore/11.2.0] Perl v5.34.0, expat v2.4.1, ...

### PR DESCRIPTION
**Pull request by Micket**
(created using `eb --new-pr`)
Instead of having duplicate perl variants, I made a "Perl.bare" and let the "Perl" bundle depend on it. 

Edit: Turns out this approach needs a minor fix in framework: We need to allow `Perl.bare` to also set the `perlver` template. Or we could just hack it into the Perl easyconfig.

### CODE REVIEW

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/697748bfefe453a870447297d024936f for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/cf1e2e3455597a510bba0f43b3924049 for a full test report.

**Comment by Micket** [comment]
Hm, so, the workaround in Perl, just avoid using `perlver`and use `version` in one place is very simple. I'm a bit surprised that the easyconfig itself doesn't resolve perlver during it's build/install steps? I see no reason why it couldn't do that, and I think it does make sense. Anyway, we don't need to allow Perl.bare to set it if we don't want.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/49d2cd5f835aadca6ab7f81210709f51 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in total)
jrlogin11.jureca - Linux centos linux 8.3.2011, x86_64, AMD EPYC 7742 64-Core Processor, Python 3.6.8
See https://gist.github.com/a3bc0995ba306f895dcd1bb3120d1b00 for a full test report.

**Comment by Micket** [comment]
Crap.. I have to specify sanity check paths manually for the perl `Bundle`...

**Comment by Micket** [comment]
Well. This seemed like a good idea, but it didn't occur to me how often the there are hardcoded assumptions in both easyblocks and easyconfigs. I.e. `--with-perl=${EBROOTPERL}/bin/perl` 
I don't know what everyone else thinks, but, I'm thinking I should just rever to the old approach and have duplicate perl installations instead, because.. ugh this kinda sucks.

**Comment by boegel** [comment]
@Micket I agree, I think splitting up the Perl installation itself as `Perl.bare` (I prefer `Perl-bare` anyway) and the Perl extensions is going to make things complicated.

Another approach could be to have `Perl` and `CPAN`, where the latter is a big bundle of Perl extensions from CPAN, that depends on `Perl`...

**Comment by Micket** [comment]
I'm considering the option of getting rid of the man pages for groff itself. Patching out makeinfo from there. I mean, who ever reads that?
That would remove the annoying circular dep, removing the need for the extra makeinfo and perl.

**Comment by Micket** [comment]
In my testing, it was enoug to just `--with-doc=no` in groff. Much nicer than having duplicate perls and makeinfo, so I really favour this approach

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/dd9572fc7d455bea31b31f5033ec9cab for a full test report.

**Comment by Micket** [comment]
@boegelbot please test @ generoso

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
jwvis00.juwels - Linux centos linux 8.3.2011, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 3.6.8
See https://gist.github.com/112c8405ae11b7c91b1552c73130d808 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
jrlogin11.jureca - Linux centos linux 8.3.2011, x86_64, AMD EPYC 7742 64-Core Processor, Python 3.6.8
See https://gist.github.com/b5d68dd1a7422b20292203e0d6747890 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
centos8-eb - Linux centos linux 8.3.2011, x86_64, Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz (skylake), Python 3.6.8
See https://gist.github.com/24b9cce2d3a9d1658da755a0dab23bbf for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
jsfl01.jusuf - Linux centos linux 7.9.2009, x86_64, AMD EPYC 7742 64-Core Processor, Python 3.6.8
See https://gist.github.com/da466d65376a48c31c84b7351b2bd717 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/d5e67e2faf58f0cad13f6c5476b94aee for a full test report.

File: easybuild/easyconfigs/p/Perl/Perl-5.34.0-GCCcore-11.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,1769 @@
...
+    }),
+    ('DBI', '1.643', {
+        'source_tmpl': 'DBI-%(version)s.tar.gz',
+        'source_urls': ['https://cpan.metacpan.org/authors/id/T/TI/TIMB'],
+        'checksums': ['8a2b993db560a2c373c174ee976a51027dd780ec766ae17620c20393d2e836fa'],
+    }),
+    ('DBD::SQLite', '1.66', {
+        'source_tmpl': 'DBD-SQLite-%(version)s.tar.gz',
+        'source_urls': ['https://cpan.metacpan.org/authors/id/I/IS/ISHIGAKI'],
+        'checksums': ['89e75dff9f66c273e0e459d51f151f88960eb7d82e2821e6902f18688ba592fe'],
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/p/Perl/Perl-5.34.0-GCCcore-11.2.0.eb:204]:
```suggestion
    ('DBD::SQLite', '1.70', {
        'source_tmpl': 'DBD-SQLite-%(version)s.tar.gz',
        'source_urls': ['https://cpan.metacpan.org/authors/id/I/IS/ISHIGAKI'],
        'checksums': ['40fd8ddf539e0e773a7a4e6d376794c3301459f9ab0050978bdcf97113dafe3e'],
```

File: easybuild/easyconfigs/p/Perl/Perl-5.34.0-GCCcore-11.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,1769 @@
...
+    }),
+    ('File::Spec', '3.75', {
+        'source_tmpl': 'PathTools-%(version)s.tar.gz',
+        'source_urls': ['https://cpan.metacpan.org/authors/id/X/XS/XSAWYERX'],
+        'checksums': ['a558503aa6b1f8c727c0073339081a77888606aa701ada1ad62dd9d8c3f945a2'],
+    }),
+    ('Test::Simple', '1.302183', {
+        'source_tmpl': 'Test-Simple-%(version)s.tar.gz',
+        'source_urls': ['https://cpan.metacpan.org/authors/id/E/EX/EXODIST'],
+        'checksums': ['9a03bda5ec420aea9692b650437f4d5b574fa505fdd7ff60cdb5f3ec034106ff'],
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/p/Perl/Perl-5.34.0-GCCcore-11.2.0.eb:234]:
```suggestion
    ('Test::Simple', '1.302186', {
        'source_tmpl': 'Test-Simple-%(version)s.tar.gz',
        'source_urls': ['https://cpan.metacpan.org/authors/id/E/EX/EXODIST'],
        'checksums': ['2895c8da7c3fe632e5714c7cc548705202cdbf3afcbc0e929bc5e6a5172265d4'],
```

**Review by SebastianAchilles — COMMENTED** [review]
Do we want to update these extensions?

**Comment by Micket** [comment]
@boegelbot please test @ generoso
EB_ARGS="--rebuild Perl-5.34.0-GCCcore-11.2.0.eb"

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/7b893f0aa542177ff28ec44bf970dec2 for a full test report.

**Review by SebastianAchilles — APPROVED** [review]
lgtm

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
centos8-eb - Linux centos linux 8.3.2011, x86_64, Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz (skylake), Python 3.6.8
See https://gist.github.com/975ef7c2b616036b3493028e35c75f31 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
jwvis00.juwels - Linux centos linux 8.3.2011, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 3.6.8
See https://gist.github.com/3febf10dd406a539c56ad6f3abb7c656 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
jsfl01.jusuf - Linux centos linux 7.9.2009, x86_64, AMD EPYC 7742 64-Core Processor, Python 3.6.8
See https://gist.github.com/6a1b54b2f3e183cbd4e49a4fd1f0bc78 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
jrlogin11.jureca - Linux centos linux 8.3.2011, x86_64, AMD EPYC 7742 64-Core Processor, Python 3.6.8
See https://gist.github.com/90d64cde74f6f9b8ec8dde0d74bee1ad for a full test report.

**Comment by SebastianAchilles** [comment]
Going in, thanks @Micket!

**Pull request merged**